src/picologging/logger.cxx

```diff
@@ -31,6 +41,10 @@ PyObject* Logger_new(PyTypeObject* type, PyObject* args, PyObject* kwds)
         Py_INCREF(self->name);
         self->parent = Py_None;
         Py_INCREF(self->parent);
+        self->children = PyList_New(0);
+        if (self->children == NULL)
+            return nullptr;
+        Py_INCREF(self->children);
```

Comment:
This additional reference isn't necessary as PyList_New returns an object with a refcnt of 1, which means the primary reference is owned by this object. With this additional reference, it would cause a memory leak as the object would still exist after Logger_dealloc is called

Reply:
Thanks for explanation, could you explain why it was needed here? filterer.cxx:

```
        self->filters = PyList_New(0);
        if (self->filters == NULL)
            return nullptr;
        Py_INCREF(self->filters);
```

Reply:
😬 hmm that might be a leak! Good spot